File: web/src/api/etf.ts
/** ETF 相关 API */
import client from './client'

export interface ETFListItem {
  code: string
  name: string
  etf_type: string | null
  nav: number | null
  market_price: number | null
  discount_rate: number | null
}

/** ETF 列表 */
export async function getETFList(params?: {
  search?: string
  etf_type?: string
  page?: number
  page_size?: number
}): Promise<{ data: ETFListItem[]; total: number; page: number; page_size: number }> {
  const { data } = await client.get('/etf', { params })
  return data
}

/** ETF 详情 */
export interface ETFDetail extends ETFListItem {
  acc_nav: number | null
  latest_date: string | null
  open: number | null
  close: number | null
  high: number | null
  low: number | null
  volume: number | null
  turnover: number | null
  pct_change: number | null
  amplitude: number | null
  turnover_rate: number | null
}

export async function getETFDetail(code: string): Promise<ETFDetail> {
  const { data } = await client.get(`/etf/${code}`)
  return data
}

/** 搜索 ETF */
export interface ETFSearchItem {
  code: string
  name: string
  etf_type: string | null
}

export async function searchETFs(keyword: string, limit = 20): Promise<ETFSearchItem[]> {
  const { data } = await client.get('/etf/search', { params: { keyword, limit } })
  return data.results
}

File: server/models/etf.py
"""ETF 相关 Pydantic schemas"""
from typing import Optional, List
from pydantic import BaseModel


class ETFListItem(BaseModel):
    """ETF 列表单项"""
    code: str
    name: str
    etf_type: Optional[str] = None
    nav: Optional[float] = None
    market_price: Optional[float] = None
    discount_rate: Optional[float] = None


class ETFDetail(ETFListItem):
    """ETF 详情"""
    acc_nav: Optional[float] = None
    latest_date: Optional[str] = None
    open: Optional[float] = None
    close: Optional[float] = None
    high: Optional[float] = None
    low: Optional[float] = None
    volume: Optional[float] = None
    turnover: Optional[float] = None
    pct_change: Optional[float] = None
    amplitude: Optional[float] = None
    turnover_rate: Optional[float] = None


class ETFListResponse(BaseModel):
    """ETF 列表响应"""
    data: List[ETFListItem]
    total: int
    page: int
    page_size: int


class ETFRealtimeItem(BaseModel):
    """ETF 实时行情单项"""
    code: str
    name: str
    nav: Optional[float] = None
    pct_change: Optional[float] = None
    fund_type: Optional[str] = None


class ETFRealtimeResponse(BaseModel):
    """ETF 实时行情响应"""
    data: List[ETFRealtimeItem]
    total: int


class ETFDailyBar(BaseModel):
    """ETF 日 K 线数据"""
    date: str
    open: Optional[float] = None
    close: Optional[float] = None
    high: Optional[float] = None
    low: Optional[float] = None
    volume: Optional[float] = None
    turnover: Optional[float] = None
    pct_change: Optional[float] = None


class ETFDailyResponse(BaseModel):
    """ETF 日 K 线响应"""
    code: str
    name: str
    data: List[ETFDailyBar]
